pretyping/unification.ml

```diff
@@ -87,6 +87,10 @@ let occur_meta_or_undefined_evar evd c =
     | _ -> Constr.iter occrec c
   in try occrec c; false with Occur | Not_found -> true
 
+let whd_meta sigma c = match EConstr.kind sigma c with
+  | Meta p -> (try meta_value sigma p with Not_found -> c)
+  | _ -> c
```

Comment:
Is it normal that this isn't recursive? ie can't you have a meta whose value is a meta?

Reply:
Do I look like I have any idea of the semantics of metas?

Reply:
Note that there is a `meta_value` function which has been cleverly given the same name (and the same type, to add insult to injury) as the one from `Evd` which *is* recursive, contrarily to the one from `Evd`. Yet, this is not the function called by `whd_meta`.